### DISCUSSION TITLE
CoAP Cipher suite support

### DISCUSSION THREAD

**Discussion by KoenR3** [emqx/emqx#13183] [Q&A] [answered]
I am trying to activate cipher suite TLS_PSK_WITH_AES_256_CCM_8 for a DTLS connection.  However, the cipher suite is not accepted.  Looking at openssl it seems to be in the psk ciphers.

I am currently testing using EMQX v5 with the default dockerfile.

Any advice if it is possible to activate the cipher suite?  I currently cannot support another one as the devices are programmed with the above cipher suite.

**Comment by zmstone** [comment, marked as answer]
HI @KoenR3 
All cipher suites supported by EMQX must be configured using OpenSSL names.
`TLS_PSK_WITH_AES_256_CCM_8` is IANA format, the OpenSSL name for it is `PSK-AES256-CCM8` [ref](https://ciphersuite.info/cs/TLS_PSK_WITH_AES_256_CCM_8/)
This cipher suite is however not supported by EMQX.

Supported cipher suites are:
```
    "RSA-PSK-AES256-GCM-SHA384",
    "RSA-PSK-AES256-CBC-SHA384",
    "RSA-PSK-AES128-GCM-SHA256",
    "RSA-PSK-AES128-CBC-SHA256",
    "RSA-PSK-AES256-CBC-SHA",
    "RSA-PSK-AES128-CBC-SHA"
```

**Reply by KoenR3** [reply]
Thank you zmstone.  I was aware of the naming but I could not find a list of supported cipher suites.

I am not super familiar with tthe ciphers but i see that these are RSA cipher suites, are these different from the TLS cipher suites?

I currently managed to configure the gateway to use the following two:
PSK-AES128-CBC-SHA256
PSK-AES128-CBC-SHA

I have not yet been able to validate a connection due to some proxy issues (DTLS support seems to be removed from nginx, so I will have to switch to another solution)
